File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
/* /dev/stdin */
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');

const [n, m] = input[0].split(' ');
let arr = [];
for(let i = 0; i < n; i++) arr.push(i + 1);
let tf = Array(m).fill(false);
let selected = [];
let answer = '';


dfs(m, 0);
console.log(answer);

function dfs(m, deps) {
  if(deps == m) {
    let temp = [];
    for(let x = 0; x < selected.length; x++) temp.push(arr[selected[x]])
    for(let x = 0; x < temp.length; x++) answer += temp[x] + ' ';
    
    answer += '\n';
    
    
  }

  for(let i = 0; i < arr.length; i++) {
    if(tf[i]) {
      
    } else {
      selected.push(i);
      tf[i] = true;
      dfs(m, deps + 1);
      selected.pop();
      tf[i] = false;
    }
  }
  
}

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
import sys

N, M = map(int,sys.stdin.readline().strip().split(' '))
arr = [str(x) for x in range(1, N + 1)]
visited = [False] * N

temp = []
def dfs(depth):
  if len(temp) == M:
    print(" ".join(temp))
    return

  for i in range(len(arr)):
    if not visited[i]:
      visited[i] = True
      temp.append(arr[i])
      dfs(depth + 1)
      temp.pop()
      visited[i] = False

dfs(0)
  

